tests/UnitTests/TemplateSource/_Issues/549/MbSplitEncodingIssue549Test.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Smarty PHPunit tests - issue #549 regression tests
+ *
+ * @package PHPunit
+ * @author  Andrey Repin <anrdaemon@yandex.ru>
+ */
+
+/**
+ * class for compiler tests
+ *
+ * @runTestsInSeparateProcess
+ * @preserveGlobalState    disabled
+ * @backupStaticAttributes enabled
+ *
+ * mb_split breaks if Smarty encoding is not the same as mbstring regex encoding.
+ */
+class MbSplitEncodingIssue549Test extends PHPUnit_Smarty
+{
+    /** @var string Saved Smarty charset */
+    private $charset;
+
+    /** @var array Source data for tests, hexed to protect from accidental reencoding */
+    private $data = array(
+        "subject" => '4772c3bc6e6577616c64', // "Grünewald"
```

Comment:
Please add "regex sensitive" characters to the test strings just to prove that your regex quoting and splitting works.

Reply:
You mean, characters like `{}` ?

Reply:
These should be added to parent tests, I presume. Since this is regression only test.

Reply:
Yes characters like that and: . + * \ / ? ( ) 
Thanks.